homeassistant/components/tellduslive/entry.py

```diff
@@ -111,3 +111,15 @@ def _last_updated(self):
     def unique_id(self) -> str:
         """Return a unique ID."""
         return self._id
+
+    @property
+    def device_info(self):
+        """Return device info."""
+        device = self.device.info()
```

Comment:
We should really change the `device` property on the tellduslive entity class to not be a property I think. It's very easy to forget that it does I/O. Properties should preferably not do I/O if not all properties of a class does that. I think we should change it so that the property returns a cached result (attribute) and we add a method that updates the attribute.

Reply:
Back in the days (a few weeks ago) I did schedule `self.device.info()` when adding a new device (around [here](https://github.com/home-assistant/home-assistant/pull/19180/files#diff-6182200c1051279a736e0dd1a80942d9R177)) and stored it in the Client. Would that be better?

Reply:
Sounds ok, if device info doesn't change.

Reply:
>  device info doesn't change.

When does Home Assistant read the new device_info?

Reply:
☺

Good point! Home assistant only reads it when adding the entity.